src/sparseml/yolov8/trainers.py

```diff
@@ -322,11 +322,58 @@ def _setup_train(self, rank, world_size):
                 epoch=self.start_epoch,
                 wrap_optim=self.scaler,
                 loggers=self.logger_manager,
+                grad_sampler={
+                    "data_loader_builder": self._get_data_loader_builder(),
+                    "loss_function": lambda preds, batch: self.criterion(preds, batch)[
+                        0
+                    ]
+                    / self.train_loader.batch_size,
+                },
             )
         else:
             # initialize steps_per_epoch for logging when there's no recipe
             self.steps_per_epoch = len(self.train_loader)
 
+    def _setup_ddp(self, rank, world_size):
+        torch.cuda.set_device(rank)
+        self.device = torch.device("cuda", rank)
+        LOGGER.info(
+            f"DDP settings: RANK {rank}, WORLD_SIZE {world_size}, DEVICE {self.device}"
+        )
+        torch.distributed.init_process_group(
+            "nccl" if torch.distributed.is_nccl_available() else "gloo",
+            rank=rank,
+            world_size=world_size,
+            timeout=timedelta(seconds=7200),
+        )
```

Comment:
Is this used anywhere?

Reply:
Yes. This is used by ultralytics to set up DDP training. The only change wrt ultralytics implementation is the `timeout` increase from 1800s to 7200s. This helps avoid timeout errors with scorer.
I will add in a comment to make this explicit.

Reply:
Ahhh gotcha, thanks!